django/utils/autoreload.py

```diff
@@ -98,7 +98,7 @@ def iter_all_python_module_files():
     # modules based on the module name and pass it to iter_modules_and_files().
     # This ensures cached results are returned in the usual case that modules
     # aren't loaded on the fly.
-    modules_view = sorted(list(sys.modules.items()), key=lambda i: i[0])
+    modules_view = sorted(sys.modules.items(), key=lambda i: i[0])
```

Comment:
@orf, is there anything performance related with this?

Reply:
I don't believe so, this does seem unneeded. As a whole, this and the two lines below are pretty performance critical to the reloader but I don't see how removing `list()` would cause any issues with that.

Reply:
`timeit` on Python 3.7 suggest that the iterator version is slightly faster:

```
$ python3 -m timeit -s 'import sys' 'sorted(list(sys.modules.items()), key=lambda i: i[0])'
20000 loops, best of 5: 10.8 usec per loop
$ python3 -m timeit -s 'import sys' 'sorted(sys.modules.items(), key=lambda i: i[0])'
20000 loops, best of 5: 10.5 usec per loop
```

Reply:
Doing a bit more testing, changing the `key` argument from to `operator.itemgetter(0)` is also slightly more performant. So, if this is a critical path, it could also be added. 

```
$ python3 -m timeit -s "
import sys
import operator
" 'sorted(sys.modules.items(), key=operator.itemgetter(0))'
50000 loops, best of 5: 8.91 usec per loop
$ python3 -m timeit -s "
import sys
import operator
" 'sorted(sys.modules.items(), key=lambda i: i[0])'
20000 loops, best of 5: 10.4 usec per loop
```

Reply:
FWIW
```
$ python3 -m timeit -s "
import sys, operator, weakref
" '
modules_view = sorted(sys.modules.items(), key=lambda i: i[0])
modules = tuple(m[1] for m in modules_view if not isinstance(m[1], weakref.ProxyTypes))
'
10000 loops, best of 5: 22.8 usec per loop

$ python3 -m timeit -s "
import sys, operator, weakref
" '
modules_view = sorted(sys.modules.items(), key=operator.itemgetter(0))
modules = tuple(m[1] for m in modules_view if not isinstance(m[1], weakref.ProxyTypes))
'
10000 loops, best of 5: 21.8 usec per loop

$ python3 -m timeit -s "
import sys, operator, weakref
" '
keys = sorted(sys.modules)
modules = tuple(m for m in map(sys.modules.__getitem__, keys) if not isinstance(m, weakref.ProxyTypes))
'
20000 loops, best of 5: 17.3 usec per loop

```